### PR TITLE
New features for repo flow

### PR DESCRIPTION
**Pull request by siggisim**
New features:
- Adds the ability to find & replace variables in templates
- Adds the ability to specify a destination directory that the template should be copied into in the new repo
- Adds the option to skip GitHub repo creation, allowing you to copy templates into existing repos
- Adds the option to skip workflow linking

I can break this out into 3-4 PRs if that's easier, but I don't think the changes are too crazy.

### CODE REVIEW

File: enterprise/server/githubapp/githubapp.go [hunk 1]
```diff
@@ -612,92 +616,136 @@ func (a *GitHubApp) CreateRepo(ctx context.Context, req *rppb.CreateRepoRequest)
...
+	gitRepo, err := initOrClone(needsInit, destDir, destURL, auth)
 	if err != nil {
 		return err
 	}
-	err = gitRepo.Storer.SetReference(plumbing.NewSymbolicReference(plumbing.HEAD, plumbing.Main))
+	fileInfo, err := ioutil.ReadDir(path)
 	if err != nil {
 		return err
 	}
+	// Copy template files into the destination repo
```
Comment (tylerwilliams) [inline enterprise/server/githubapp/githubapp.go:738]: nit: os.Rename does move (not copy), right?

File: enterprise/server/githubapp/githubapp.go [hunk 2]
```diff
@@ -713,22 +761,53 @@ func cloneTemplate(email, tmpDirName, token, srcURL, destURL, srcDir string) err
...
+		return nil, err
+	}
+	_, err = gitRepo.CreateRemote(&config.RemoteConfig{
+		Name: git.DefaultRemoteName,
+		URLs: []string{url},
+	})
+	return gitRepo, err
+}
+
+func replace(dir string, replacements map[string]string) error {
```
Comment (tylerwilliams) [inline enterprise/server/githubapp/githubapp.go:792]: nit: add a comment

**Review by tylerwilliams — APPROVED** [review]

File: enterprise/server/githubapp/githubapp.go (hunk 2 above)
Comment (siggisim) [reply enterprise/server/githubapp/githubapp.go:794]: Done

File: enterprise/server/githubapp/githubapp.go (hunk 1 above)
Comment (siggisim) [reply enterprise/server/githubapp/githubapp.go]: Fixed

**Pull request merged**